gateway/src/integration-test/java/com/hurence/webapiservice/http/grafana/AnnotationEndPointIT.java

```diff
@@ -0,0 +1,186 @@
+    public void testAnnotationWithNoTime(Vertx vertx, VertxTestContext testContext) {
+        assertRequestGiveObjectResponseFromFileWithNoOrder(vertx, testContext,
+                "/http/grafana/annotation/test6/request.json",
+                "/http/grafana/annotation/test6/expectedResponse.json");
+    }
+
+    @Test
+    @Timeout(value = 5, timeUnit = TimeUnit.SECONDS)
+    public void testAnnotationWithEmptyQuery(Vertx vertx, VertxTestContext testContext) {
+        assertRequestGiveObjectResponseFromFile(vertx, testContext,
+                "/http/grafana/annotation/test5/request.json",
+                "/http/grafana/annotation/test5/expectedResponse.json");
+    }
+
+
+    public void assertRequestGiveObjectResponseFromFile(Vertx vertx, VertxTestContext testContext,
+                                                  String requestFile, String responseFile) {
+        assertHelper.assertRequestGiveObjectResponseFromFile(vertx, testContext, requestFile, responseFile);
+    }
+
+    public void assertRequestGiveObjectResponseFromFileWithNoOrder(Vertx vertx, VertxTestContext testContext,
+                                                                   String requestFile, String responseFile) {
+        final FileSystem fs = vertx.fileSystem();
+        Buffer requestBuffer = fs.readFileBlocking(AssertResponseGivenRequestHelper.class.getResource(requestFile).getFile());
+        webClient.post("/api/grafana/annotations")
+                .as(BodyCodec.jsonObject())
+                .sendBuffer(requestBuffer.getDelegate(), testContext.succeeding(rsp -> {
+                    testContext.verify(() -> {
+                        assertEquals(200, rsp.statusCode());
+                        assertEquals("OK", rsp.statusMessage());
+                        JsonObject body = rsp.body();
+                        Buffer fileContent = fs.readFileBlocking(AssertResponseGivenRequestHelper.class.getResource(responseFile).getFile());
+                        JsonObject expectedBody = new JsonObject(fileContent.getDelegate());
+                        int totalExpected = expectedBody.getInteger(RESPONSE_TOTAL_FOUND);
+                        int total = body.getInteger(RESPONSE_TOTAL_FOUND);
+                        JsonArray annotationExpected = expectedBody.getJsonArray(RESPONSE_ANNOTATIONS);
+                        JsonArray annotations = body.getJsonArray(RESPONSE_ANNOTATIONS);
+                        boolean isEqual = true;
+                        for (Object je1 : annotations) {
+                            boolean flag = false;
+                            for(Object je2 : annotationExpected){
+                                flag = je1.equals(je2);
+                                if(flag){
+                                    break;
+                                }
+                            }
+                            isEqual = isEqual && flag;
+                        }
+                        assertTrue(isEqual);
```

Comment:
I do not think this code does what you think it does. It will iterate over "annotations" and once ones of the annotation equals one on the expected ones. It will set isEqual to true. 
That's means your test just verify that annotations contains at least one annotation among the expected ones.

Reply:
If you do not care about order. Maybe you could try to use collections that does not care about order. Like Set (https://docs.oracle.com/javase/8/docs/api/java/util/HashSet.html) for example :)

Reply:
yes you're right ! it doesn't do what i think, i will use Set